File: examples/hybridsearch/HybridSearchCSharp/OobDev.Search.OpenSearch/OpenSearchOptions.cs
namespace OobDev.Search.OpenSearch;

public class OpenSearchOptions
{
    public required string HostName { get; set; } = "localhost";
    public required int Port { get; set; } = 9200;
    public required string IndexName { get; set; } = "default";

    public string? UserName { get; set; }
    public string? Password { get; set; }
}

File: examples/hybridsearch/HybridSearchGo/providers/lexical/OpenSearchOptions.go
package lexical

import (
	"fmt"
)

type OpenSearchOptions struct {
	HostName  string
	Port      int
	IndexName string

	UserName string
	Password string
}

func CreateOptions() OpenSearchOptions {
	return OpenSearchOptions{
		HostName:  "192.168.1.170",
		Port:      9200,
		IndexName: "docs",
	}
}

func (opt OpenSearchOptions) Url() string {
	return fmt.Sprintf("http://%s:%v", opt.HostName, opt.Port)
}
